dvc/repo/imp.py

```diff
@@ -1,13 +1,28 @@
+import os
+
+from dvc.utils.compat import urlparse
+
 from dvc.repo.scm_context import scm_context
+from dvc.output import OutputBASE
+from dvc.output import loads_from as load_outs
+from dvc.dependency import loads_from as load_deps
 
 
 @scm_context
-def imp(self, url, out, resume=False, fname=None):
+def imp(self, url, out=None, resume=False, fname=None):
     from dvc.stage import Stage
 
+    deps = load_deps([url])
+
+    if not out:
+        dep_url = deps[0][OutputBASE.PARAM_PATH]
+        out = os.path.basename(urlparse(dep_url).path)
```

Comment:
so, out is optional now, right? let's create a chekbox in the doc ticket for packages to specify the difference please

Reply:
It has been optional always. Just moved some logic from CLI to API.